model/edition.py

```diff
@@ -225,6 +229,32 @@ def dedupe(l):
             return []
 
 
+    @classmethod
+    def medium_from_media_type(cls, media_type):
+        """Derive a value for Edition.medium from a media type.
+
+        TODO: It's not necessary right now, but we could theoretically
+        derive this information from some other types such as
+        our internal types for Overdrive manifests.
+
+        :param media_type: A media type with optional parameters
+        :return: A value for Edition.medium.
+        """
+        if not media_type:
+            return None
+
+        for types, conclusion in (
+            (MediaTypes.AUDIOBOOK_MEDIA_TYPES, Edition.AUDIO_MEDIUM),
+            (MediaTypes.BOOK_MEDIA_TYPES, Edition.BOOK_MEDIUM),
+        ):
+            if any(media_type.startswith(x) for x in types):
+                return conclusion
+
+        if media_type.startswith(DeliveryMechanism.ADOBE_DRM):
+            # Adobe DRM is only applied to ebooks.
+            return Edition.BOOK_MEDIUM
+        return None
+
```

Comment:
This method contains code refactored from `OPDSImporter.get_medium_from_links` (so we can also call it on the contents of a `<dcterms:format>` tag) and greatly expanded.